source/main.cpp

```diff
@@ -27,13 +27,14 @@ int main(int argc, char** argv)
     */
     int nproc = 1;
     int my_rank = 0;
-    Parallel_Global::read_mpi_parameters(argc, argv, nproc, my_rank);
+    int nthread_per_proc = 1;
+    Parallel_Global::read_mpi_parameters(argc, argv, nproc, nthread_per_proc, my_rank);
```

Comment:
now the omp information is also got here, so it is not appropriate to let this function still have name `read_mpi_parameters`, instead, I recommend `read_pal_param`